test/gitea_test.go

```diff
@@ -577,6 +577,32 @@ func TestGiteaWithCLIGeneratePipeline(t *testing.T) {
 	}
 }
 
+func TestGiteaCancelRun(t *testing.T) {
+	topts := &tgitea.TestOpts{
+		TargetEvent: options.PullRequestEvent,
+		YAMLFiles: map[string]string{
+			".tekton/pr.yaml": "testdata/pipelinerun_long_running.yaml",
+		},
+		NoCleanup:    true,
+		ExpectEvents: false,
+	}
+	defer tgitea.TestPR(t, topts)()
+
+	// let pipelineRun start and then cancel it
+	time.Sleep(time.Second * 5)
+	tgitea.PostCommentOnPullRequest(t, topts, "/cancel")
+
+	waitOpts := twait.Opts{
+		RepoName:        topts.TargetNS,
+		Namespace:       topts.TargetNS,
+		MinNumberStatus: 1,
+		PollTimeout:     twait.DefaultTimeout,
+		TargetSHA:       topts.PullRequest.Head.Sha,
+	}
+	err := twait.UntilRepositoryUpdated(context.Background(), topts.Clients.Clients, waitOpts)
```

Comment:
you can inspect the event of namespace to see if there was a message attached to the repo CR (there is other examples that does this in the tests) and see if it was canceled... the untilrepoupdated is not great and very error prone...

Reply:
checking pipelinerun directly